src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -274,29 +275,29 @@ private GenericInfoBucketData createGenericInfoBucketData(BucketEntity bucket) {
         return result.stream().map(CatalogObjectMetadata::new).collect(Collectors.toList());
     }
 
-    public List<CatalogObjectMetadata> listCatalogObjectsByKind(String bucketName, String kind) {
-        findBucketByNameAndCheck(bucketName);
-        List<CatalogObjectRevisionEntity> result = catalogObjectRevisionRepository.findDefaultCatalogObjectsOfKindInBucket(bucketName,
+    public List<CatalogObjectMetadata> listCatalogObjectsByKind(List<String> bucketNames, String kind) {
+        bucketNames.forEach(bucketName -> findBucketByNameAndCheck(bucketName));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)